instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpServerTracer.java

```diff
@@ -58,67 +69,51 @@ public Context startSpan(
     onRequest(span, request);
     onConnectionAndRequest(span, connection, request);
 
-    return parentContext.with(span);
-  }
+    Context context = parentContext.with(CONTEXT_SERVER_SPAN_KEY, span).with(span);
+    attachServerContext(context, storage);
 
-  /**
-   * Creates new scoped context, based on the current context, with the given span.
-   *
-   * <p>Attaches new context to the request to avoid creating duplicate server spans.
-   */
-  public Scope startScope(Span span, STORAGE storage) {
-    return startScope(span, storage, Context.current());
-  }
-
-  /**
-   * Creates new scoped context, based on the given context, with the given span.
-   *
-   * <p>Attaches new context to the request to avoid creating duplicate server spans.
-   */
-  public Scope startScope(Span span, STORAGE storage, Context context) {
-    // TODO we could do this in one go, but TracingContextUtils.CONTEXT_SPAN_KEY is private
-    Context newContext = context.with(CONTEXT_SERVER_SPAN_KEY, span).with(span);
-    attachServerContext(newContext, storage);
-    return newContext.makeCurrent();
```

Comment:
I don't understand where this happens now. We have to active new context somewhere, haven't we?

Reply:
A, I see, every instrumentation now does that itself.

I have one problem with this and with the removal of `startScope`: inconsistency. Now instrumentations which use `HttpServerTracer` and those which use e.g. `HttpClientTracer` are different. They used to follow the same pattern: `startSpan+startScope`, now they are going to be different. Maybe it is worth leaving `startScope` method here with just `context.makeCurrent()` to preserve uniformity?

Reply:
#1811 😁